src/extension/background-script/nostr/index.ts

```diff
@@ -69,6 +82,18 @@ class Nostr {
     return Utf8.stringify(decrypted);
   }
 
+  nip44Encrypt(pairs: [string, string][]) {
+    return pairs.map(([pubkey, text]: [string, string]) =>
+      nip44.v2.encrypt(text, this.getNip44SharedSecret(pubkey))
+    );
+  }
+
+  nip44Decrypt(pairs: [string, string][]) {
+    return pairs.map(([pubkey, payload]) =>
+      nip44.v2.decrypt(payload, this.getNip44SharedSecret(pubkey))
+    );
+  }
```

Comment:
Per https://github.com/nostr-protocol/nips/pull/940#discussion_r1496216981, these functions throw if any one element in them is invalid, making it infeasible for practical use, because one invalid message can DoS a client from decrypting all messages.

Reply:
I'll update this once we know what we're doing with nip 07

Reply:
They don't have to throw if any one element is invalid, you can specify that the array of responses can contain either results or error values in it.

Reply:
@fiatjaf That was already suggested here: https://github.com/nostr-protocol/nips/pull/940#discussion_r1495190631

Yes it could solve the problem. But why would we prefer that over atomic calls?

Reply:
I like atomic calls more, but you're mangling the atomic calls with a breaking change in NIP-46 for "batched requests", and that feels horribly wrong to me.

I would rather stop and come up with a double-ratcheting scheme that is resistant to leaked conversation keys and use simple calls for apps to just get temporary conversation keys or something like that.

Reply:
> you're mangling the atomic calls with a breaking change in NIP-46 for "batched requests", and that feels horribly wrong to me.

I completely agree, that is the whole argument I've been trying to make. People are saying NIP-46 and NIP-07 need to match identically. I'm trying to show people that they can still be compatible even though they have different interfaces.

Reply:
As far as I know no one has implemented NIP-44 stuff in either NIP-46 or NIP-07, so we can come up with whatever interface we want.

But then we go and we break the normal `signEvent()` flow that already works? Why?

Reply:
> no one has implemented NIP-44 stuff in either NIP-46 or NIP-07, so we can come up with whatever interface we want.

I agree, so please approve https://github.com/nostr-protocol/nips/pull/1063